### PR TITLE
Check for frees

### PR DESCRIPTION
**Pull request by HosseinYousefi**
As discussed in #359, I removed `ensureNotDeleted` and instead included it in the `reference` getter. The previous tests testing for exceptions, will test this change.

Also changing the pubspec version to fix the beta CI.

### CODE REVIEW

File: jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// This is a generated file; do not edit or check into version control.
```
Comment (dcharkes) [inline jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig:1]: Looks like we're missing a gititgnore somewhere.

File: jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/flutter_export_environment.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (dcharkes) [inline jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/flutter_export_environment.sh:2]: ditto

File: jni/lib/src/jexceptions.dart [hunk 3]
```diff
@@ -15,21 +15,21 @@ class UseAfterFreeException implements JException {
...
   String toString() {
-    return "use after free on $ptr through $object";
+    return "Use after free on $ptr through $object.";
   }
 }
 
 class NullJStringException implements JException {
   @override
-  String toString() => 'toDartString called on null JString reference';
+  String toString() => 'toDartString was called on null JString reference.';
```
Comment (dcharkes) [inline jni/lib/src/jexceptions.dart:24]: Maybe put `toDartString` in backticks or quotes.

**Review by dcharkes — APPROVED** [review]

File: jni/lib/src/jreference.dart [hunk 4]
```diff
@@ -27,31 +23,41 @@ extension ProtectedJReference on JReference {
...
-    _finalizer.attach(this, reference, detach: this);
+  JReference.fromRef(this._reference) {
+    _finalizer.attach(this, _reference, detach: this);
   }
 
   bool _deleted = false;
 
   /// Check whether the underlying JNI reference is `null`.
-  bool get isNull => reference == nullptr;
+  bool get isNull => _reference == nullptr;
```
Comment (HosseinYousefi) [inline jni/lib/src/jreference.dart:36]: Should this be `reference` or `_reference`?

Comment (HosseinYousefi) [reply jni/lib/src/jreference.dart:36]: @dcharkes

File: jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig (hunk 1 above)
Comment (HosseinYousefi) [reply jnigen/example/pdfbox_plugin/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig:1]: macos is not supported yet, so I'll just remove it. It was accidentally added after the pub version change + upgrade.

File: jni/lib/src/jreference.dart (hunk 4 above)
Comment (HosseinYousefi) [reply jni/lib/src/jreference.dart:36]: I guess it doesn't really matter to check it here as well, so I'll go with `_reference`.

Comment (dcharkes) [reply jni/lib/src/jreference.dart:36]: Looking from the tests, the uses actually are null (for example in the list test). If we don't want these references to change value over time (also not by deleting), I'd say it should be `reference`.

**Pull request merged**